src/SM/StateMachine/StateMachine.php

```diff
@@ -212,20 +213,23 @@ protected function setState($state)
      * Builds and calls the defined callbacks
      *
      * @param TransitionEvent $event
-     * @param string          $position
+     * @param string $position
+     * @return bool
      */
     protected function callCallbacks(TransitionEvent $event, $position)
     {
+        $result = true;
         if (!isset($this->config['callbacks'][$position])) {
-            return;
+            return $result;
         }
 
         foreach ($this->config['callbacks'][$position] as &$callback) {
             if (!$callback instanceof CallbackInterface) {
                 $callback = $this->callbackFactory->get($callback);
             }
 
-            call_user_func($callback, $event);
+            $result = $result & call_user_func($callback, $event);
```

Comment:
This is no typo at all: when using `&&`, the first callback that returns `false` breaks the execution chain, so that no other callbacks are executed.

This is necessarily `&` in order to not break existing `before` and `after` handlers, which do not return a boolean at all, thus returning `null` and evaluating to `false`.

Another option would be to switch the operands:
```
$result = call_user_func($callback, $event) && $result;
```

What do you think about it?